cpp/velox/memory/VeloxMemoryManager.cc

```diff
@@ -168,7 +168,13 @@ class ListenableArbitrator : public velox::memory::MemoryArbitrator {
     }
     auto reclaimedFreeBytes = shrinkPool(pool, 0);
     auto neededBytes = velox::bits::roundUp(bytes - reclaimedFreeBytes, memoryPoolTransferCapacity_);
-    listener_->allocationChanged(neededBytes);
+    try {
+      listener_->allocationChanged(neededBytes);
+    } catch (const std::exception& e) {
+      // if allocationChanged failed, we need to free the reclaimed bytes
+      listener_->allocationChanged(-reclaimedFreeBytes);
+      throw e;
```

Comment:
We should use `throw;` instead of `throw e;`. Otherwise the original exception type will be cast down to `std::exception`.

Reply:
> We should use `throw;` instead of `throw e;`. 

Thank you, I will send a follow up PR to change it

> Otherwise the original exception type will be cast down to `std::exception`.

Sorry, I'm not familiar with c++, could this cause any problems? how can I handle it?

Reply:
Is it ok like this?

```
    try {
      listener_->allocationChanged(neededBytes);
    } catch (const std::exception&) {
      // if allocationChanged failed, we need to free the reclaimed bytes
      listener_->allocationChanged(-reclaimedFreeBytes);
      std::rethrow_exception(std::current_exception());
    }
```